File: multicore/demo-mul.rb
#!/usr/bin/ruby
require 'digest/md5'

def fexe(value="")
    digest = Digest::MD5.hexdigest(value)

    for i in 1..10000
        digest = Digest::MD5.hexdigest(value)
    end
end

count = 0
arr = []

msg = "http://php.net/manual/en/function.printf.php";

for i in 1..10000    
   arr[i-1] = Thread.new {
      fexe(msg)
   } 
end

puts "size: #{arr.size}"

arr.each {|t| t.join; }
puts "done!"

File: multicore/demo-mul.go
package main

import (
    //"fmt"
    //"time"
    "crypto/md5"
    // "fmt"
    // "io"
)

func fexe(s string) {
    data := []byte("http://php.net/manual/en/function.printf.php")
    md5.Sum(data)

    for i := 0; i < 10000; i++ {
        md5.Sum(data)
    }
}

func main() { 
    for i := 1; i < 10000; i++ {
        go fexe("none")
    }
    fexe("none")
}